pkg/datapath/linux/probes/probes.go

```diff
@@ -277,9 +249,9 @@ func (p *ProbeManager) GetOptionalConfig() map[KernelParam]bool {
 	config := p.features.SystemConfig
 	kernelParams := make(map[KernelParam]bool)
 
-	kernelParams["CONFIG_CGROUP_BPF"] = config.ConfigCgroupBpf.Enabled()
+	kernelParams["CONFIG_CGROUP_BPF [Host Reachable Services/sockmap optimization]"] = config.ConfigCgroupBpf.Enabled()
```

Comment:
Please be consistent on case (title case nowhere or (I would prefer) everywhere). In any case, do use uppercase letters at the beginning of proper nouns.

Also, there descriptions should absolutely **not** be part of the key in `kernelParams`. It will make it very difficult to access a given element in the map if we want to check a specific parameter, or even if we want to use the parameter name in any way.

I think the ideal solution here would be to turn `kernelParam` into a `map[KernelParam]kernelOption` (or some similar name instead of `kernelOption`), where `kernelOption` is a `struct` containing the boolean status (parameter enabled/disabled), a description of the features it is used for (can be an empty string for required parameters), and then possibly another bool to tell if the option can be a module (so we could avoid the `switch`/`case` in `SystemConfigProbes()`). What do you think?

Reply:
> I think the ideal solution here would be to turn `kernelParam` into a `map[KernelParam]kernelOption` (or some similar name instead of `kernelOption`), where `kernelOption` is a `struct` containing the boolean status (parameter enabled/disabled), a description of the features it is used for (can be an empty string for required parameters), and then possibly another bool to tell if the option can be a module (so we could avoid the `switch`/`case` in `SystemConfigProbes()`). What do you think?

@qmonnet thanks for your input. so you mean just have a `struct` called `kernelOption`, and this struct is in turn incpororating `Enabled()`, so we can replace e.g.  

`kernelParams["CONFIG_CGROUP_BPF"] = config.ConfigCgroupBpf.Enabled()`

into something like 

`kernelParams["CONFIG_CGROUP_BPF"] = kernelOption`, having a map `map[KernelParam]kernelOption` like you've mentioned above?

Reply:
Yes, something like this (not tested):

```go
kernelParams["CONFIG_CGROUP_BPF"] = kernelOption{
        Enabled: config.ConfigCgroupBpf.Enabled(),
        Description: "Host Reachable Services and Sockmap optimization",
        CanBeModule: false,
}
```

Reply:
@qmonnet thanks. But if we do it that way, dumping the `JSON` ouptut of `kernel-check` is going to fail as it's expecting a `bool`  - but acutally is getting a `struct` literal. 

`map[KernelParam]bool` vs `map[KernelParam]kernelOption`

`probes.go` 

```
[... ]
type kernelOption struct {
	Description string
	Enabled     bool
	CanBeModule bool
}

[...] 
func (p *ProbeManager) GetRequiredConfig() map[KernelParam]kernelOption {
	config := p.features.SystemConfig
	kernelParams := make(map[KernelParam]kernelOption)

	kernelParams["CONFIG_BPF"] = kernelOption{
		Enabled:     config.ConfigBpf.Enabled(),
		Description: "",
		CanBeModule: false,
	}
[...]

```

`kernel_check.go`

```
type KernelParamSupport struct {
	Feature   probes.KernelParam
	Supported bool 
	Required  bool
}

```

```
func kernelCheck() {
	probeManager := probes.NewProbeManager()
	listParams := []KernelParamSupport{}
	requiredParams := probeManager.GetRequiredConfig()
	for f, s := range requiredParams {
		listParams = append(listParams, KernelParamSupport{
			Feature:   f,
			Supported: s, // failing 
			Required:  true,
		})
```

Do you may have any ideas on how to circumvent this?

Reply:
Sorry I don't understand the problem, it should be trivial to update `kernelCheck` to pass the boolean value? Something like `Supported: s.Enabled`, probably?